### DISCUSSION TITLE
Dockerfile for the kippo honeypot

### DISCUSSION THREAD

**Discussion by aristofanischionis** [telekom-security/tpotce#1373] [Ideas]
**Is your feature request related to a problem? Please describe.**
I needed to run kippo on my local machine, in order to test my tool, [Honeyscanner: a vulnerability analyzer for honeypots](https://github.com/honeynet/honeyscanner), and test its efficiency against kippo.

**Describe the solution you'd like**
I create a Dockerfile that allows anyone to easily run kippo, on any machine that supports Docker. You can find all the information here: https://github.com/aristofanischionis/kippo_Dockerfile/tree/main.

If you just want the **Dockerfile**:

```Dockerfile
# Use an official Python runtime as a parent image
FROM python:2.7-slim

# Create a non-root user and switch to it
RUN adduser --disabled-password --gecos '' app
USER app

# Set the working directory in the container to /app
WORKDIR /app

# Copy the current directory contents into the container at /app
COPY . /app

# Switch back to root to install dependencies
USER root

# Install necessary packages and clean up
RUN apt-get update && apt-get install -y \
    python-dev=2.7.* \
    build-essential=* \
    mariadb-client=* \
    libmariadbclient-dev=* && \
    rm -rf /var/lib/apt/lists/*

# Create the data and log directory and set their permissions
RUN mkdir -p /app/data /app/log /app/log/tty && \
    chown -R 777 /app/log/tty && \
    chown -R app:app /app/data /app/log

# Install Python packages
RUN pip install --no-cache-dir zope.interface==5.5.2 Twisted==15.1.0 pycrypto==2.6.1 pyasn1==0.5.0

# Switch back to the non-root user
USER app

# Make port 2222 available to the world outside thiscontainer
EXPOSE 2222

# Run twistd command when the container launches
CMD ["twistd", "-n", "-y", "kippo.tac", "-l", "log/kippo.log", "--pidfile", "kippo.pid"]
```

If you just want the **Docker image**: https://hub.docker.com/r/aristofanischionis/kippo.

**Comment by t3chn0m4g3** [comment]
@aristofanischionis We do not use Kippo, only its successor Cowrie. You may find `docker-compose.yml` files for all honeypots in the `docker` folder.
Other than that I also do not really understand the purpose of this issue😇.

**Comment by aristofanischionis** [comment]
Hi @t3chn0m4g3,
the idea to create this Dockerfile came when I was trying to run kippo for a research project. I found it hard to run it on my local machine as it requires very old and outdated libraries and packages. Apart from solving this problem for myself, I thought it could benefit others in similar situations. If you don't want it, we can just close the issue, no problem.
Thank you for your time and effort to review this.
